python: Propagate packageOverrides to pythonForBuild

adisbladis opened the pull request:
When overriding versions of build tools injected via hooks by `pythonForBuild`
`packageOverrides` was not taken into account and 2 incompatible
versions of the same package (`wheel` in this case) ended up in the
closure, causing the builds to fail.

This was broken by b57c5d44569e236edbc142c0f95772fca2bb6dad.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

FRidh commented:
As follow-up of https://github.com/NixOS/nixpkgs/pull/104201 I think it would be good if we get rid of passing around `pythonForBuild` explicitly but instead have `pythonPackagesBuildHost`, for uniformity. As far as I can see that would solve this issue.

adisbladis committed:
python: Propagate packageOverrides to pythonForBuild

When overriding versions of build tools injected via hooks
`packageOverrides` was not taken into account and 2 incompatible
versions of the same package (`wheel` in this case) ended up in the
closure, causing the builds to fail.

FRidh commented:
Have you tested again *without* this change on master?

adisbladis commented:
@FRidh Yes I have.

FRidh commented:
Alright. It would be nice to communicate that before merging this yourself.

`.overrideScope'` still led to the same issue?

@Ericson2314 seems like, while splicing is now fixed, overrides are not yet taken into account.

We'll need to make a test for this feature.
